### PR TITLE
feat: render ACP embedded resource content blocks in WeChat

### LINKED ISSUE #59: Feature: render ACP embedded resource content blocks in WeChat
## Problem

When the ACP agent streams an embedded resource content block (`type: "resource"`), wechat-acp drops it silently. The user sees the surrounding text but never learns the resource existed.

Per the ACP spec ([Embedded Resource](https://agentclientprotocol.com/protocol/v1/content#embedded-resource)), the block carries complete resource contents inline:

```jsonc
{
  "type": "resource",
  "resource": {
    "uri": "file:///home/user/script.py",
    "mimeType": "text/x-python",
    "text": "def hello():\n    print('Hello, world!')"
  }
}
```

`resource` is either `TextResourceContents` (`uri` + `text` + optional `mimeType`) or `BlobResourceContents` (`uri` + base64 `blob` + optional `mimeType`). Agents use these to hand back file contents from tool results, and the same block can appear directly as `agent_message_chunk` content or wrapped in a `ToolCallContent` of `type: "content"` inside `tool_call_update`.

Note the asymmetry: the inbound adapter (`src/adapter/inbound.ts`) already produces `type: "resource"` blocks when a WeChat user sends a text file to the agent. The outbound direction ignores the same block type entirely.

## Current behavior

In `src/acp/client.ts` `handleSessionUpdate()`:

- `agent_message_chunk` handles `text` and `image`; `resource` falls through.
- `tool_call_update` completed content handles `diff` and `content`/`image`; `content`/`resource` falls through.

## Proposal

Render embedded resources by content kind:

1) **Text resources** (`resource.text`): append to the text stream as a fenced code block with a one-line header naming the resource, so it flushes in order with the surrounding narrative through the existing buffer machinery:

   ```
   📎 script.py (text/x-python)
   ```python
   def hello():
       print('Hello, world!')
   ```
   ```

   - Header shows the last path segment of `uri` (fall back to the full URI when parsing fails) and the `mimeType` when present.
   - Fence language derived from the MIME type or file extension where obvious; plain fence otherwise. Inner content containing triple backticks gets a longer fence so the block cannot break out.
   - Cap inline rendering at 4000 characters (`TEXT_CHUNK_LIMIT`, one WeChat segment); beyond that, truncate with a `... [truncated, N more chars]` tail line inside the fence. Resources are context payloads; a WeChat chat is not the place to scroll a 200 KB file, and the cap keeps one resource from flooding a turn with dozens of segments.

2) **Blob resources with an image MIME type** (`resource.blob` + `mimeType: image/*`): route through the existing image delivery pipeline from #52 (`maybeSendImage`), so a PNG handed back as an embedded resource renders as a native WeChat image message with the same allow-list, size cap, and failure placeholder semantics.

3) **Other blob resources** (binary, no rendering path): append a one-line placeholder to the text stream instead of dropping silently:

   ```
   📎 [resource: data.bin (application/octet-stream, ~12345 bytes) - binary content not rendered]
   ```

   Byte size estimated from the base64 length. Delivering these as WeChat file messages is a natural follow-up once an outbound file-message primitive exists (the audio issue proposes one), but is intentionally out of scope here so the two changes stay independent.

### Behavior notes

- Handling runs in the same serialized per-client task queue with per-turn `TurnState` binding (issue #54); text renders buffer into `turn.chunks` at the position the block arrived, so ordering versus surrounding text is preserved for free.
- A resource-bearing turn counts as produced output (sets `producedMessage`), so a resource-only reply does not trigger the empty-turn notice.
- Config: `agent.showResources` (default `true`) plus `--hide-resources`, mirroring `showImages` / `--hide-images`. When off, resources are logged and skipped.
- Empty text resources (blank `text`) are logged and skipped rather than rendering an empty fence.

### Files touched (implementation sketch)

- `src/acp/client.ts`: `resource` block detection in both notification paths, text/blob dispatch, fence rendering with cap, image-blob routing into `maybeSendImage`, `showResources` gate.
- `src/config.ts` + `bin/wechat-acp.ts`: `showResources` config and `--hide-resources` flag.
- `src/acp/session.ts` / `src/bridge.ts`: no new callbacks needed; text renders ride `onMessageFlush`, image blobs ride the existing `onImageFlush`.
- Tests: text resource rendered fenced and in order, header/fallback naming, truncation, backtick-escape fence, image blob delivered through the image pipeline exactly once, non-image blob placeholder, flag off, empty text skipped, turn binding.

Related: #52 (images, merged in 0.9.0). I have a working implementation and will open a PR shortly.

### PR DESCRIPTION
**Pull request by Eskibear**
Fixes #59

## Problem

ACP embedded resource content blocks (`{type: "resource", resource: {uri, text | blob, mimeType?}}`) streamed in `agent_message_chunk` or in completed `tool_call_update` content are silently dropped. The user sees the surrounding narrative but never learns the resource existed. The asymmetry is worth noting: the inbound adapter already produces `type: "resource"` blocks when a WeChat user sends a text file to the agent; the outbound direction ignored the same block type entirely.

## Fix

Render by content kind, exactly as specced in issue #59:

- **Text resources** append to the text stream as a fenced code block with a `📎 <name> (<mimeType>)` header line, so they flush in stream order with the surrounding text through the existing buffer machinery (no new callbacks, no new send paths). The name is the last path segment of the URI, percent-decoded, falling back to the full URI when there is no useful segment. The fence language derives from the MIME type first, then the file extension. The fence is always longer than any backtick run inside the body, so resource content cannot break out of the block. Bodies above 4000 characters (one WeChat segment) truncate with an explicit `... [truncated, N more chars]` tail inside the fence.
- **Blob resources with an `image/*` MIME type** route through the existing image pipeline from issue #52 (`maybeSendImage`), inheriting the allow-list, the 10 MiB cap, text-before-image ordering, and both failure placeholders.
- **Other blob resources** surface as a one-line `📎 [resource: <name> (<mime>, ~N bytes) - binary content not rendered]` placeholder instead of vanishing. Delivering these as WeChat file messages is intentionally out of scope; it layers on the outbound file-message primitive proposed in issue #58, keeping the two changes independent.
- Empty text resources are logged and skipped rather than rendering an empty fence.
- Config/CLI: `agent.showResources` (default `true`), `--hide-resources`, mirroring the image flags.

Everything runs inside the serialized per-client task queue with per-turn state binding (issue #54 semantics), and a resource-bearing turn sets `producedMessage`, so a resource-only reply does not trigger the empty-turn notice. The MIME and extension lookup tables are gated with `Object.hasOwn`, so prototype-chain keys cannot resolve to a language.

## Tests

11 new tests, 48 total, all passing.

`tests/client.test.ts`: fenced render with header from both streaming paths, stream-order interleaving with surrounding text in a single flush segment, URI fallback naming, extension-based language fallback, truncation tail with segment-size bound, longer-fence escape for bodies containing triple backticks, image blob delivered through the image sink exactly once with data intact, non-image blob placeholder shape, missing-MIME blob reported as unknown type, `showResources=false` dropping both text and image-blob resources, and empty text skipped.

Verification: `npm run build` clean, full suite 48/48 green.

Note for the merge order: this PR and the audio PR (issue #58) both touch the content dispatch in `client.ts` and the config/flag tables; whichever lands second will need a trivial rebase, which I will handle.

### CODE REVIEW

File: src/acp/client.ts [hunk 1]
```diff
@@ -29,6 +29,124 @@ const SUPPORTED_IMAGE_MIME_TYPES = new Set([
...
+  const segments = stripped.split("/").filter((s) => s.length > 0);
+  const last = segments[segments.length - 1] ?? "";
+  // A lone scheme-bearing segment is not a useful name.
+  if (!last || (segments.length === 1 && last.includes(":"))) return trimmed;
+  try {
+    return decodeURIComponent(last);
+  } catch {
+    return last;
+  }
+}
```
[the remark this answers is not in the document]
Comment (Eskibear) [reply src/acp/client.ts:154]: Valid, fixed in 0513405. `resourceDisplayName` percent-decodes the segment, so `%0A` did come back as a literal newline. All return paths now go through a shared `sanitizeInlineLabel` (control chars and Unicode line separators collapsed to spaces, whitespace collapsed, 120-char cap so a crafted URI cannot flood the header either), with a final `"resource"` fallback when nothing printable survives. Regression tests cover encoded LF/CR/ESC in the URI and an overlong name.

File: src/acp/client.ts [hunk 2]
```diff
@@ -29,6 +29,124 @@ const SUPPORTED_IMAGE_MIME_TYPES = new Set([
...
+      text.slice(0, MAX_RESOURCE_TEXT_CHARS) +
+      `\n... [truncated, ${text.length - MAX_RESOURCE_TEXT_CHARS} more chars]`;
+  }
+  body = body.replace(/\n$/, "");
+  const longestRun = body.match(/`+/g)?.reduce((max, run) => Math.max(max, run.length), 0) ?? 0;
+  const fence = "`".repeat(Math.max(3, longestRun + 1));
+  const lang = resourceFenceLanguage(name, mimeType);
+  const mime = mimeType?.trim();
+  const mimeNote = mime ? ` (${mime})` : "";
+  return `\n📎 ${name}${mimeNote}\n${fence}${lang}\n${body}\n${fence}\n`;
```
[the remark this answers is not in the document]
Comment (Eskibear) [reply src/acp/client.ts]: Valid, fixed in 0513405. The header mime note now uses `sanitizeInlineLabel(mimeType)` instead of a bare trim, same sanitizer as the display name. Regression test asserts `text/plain\nFAKE: injected` renders as a single-line `(text/plain FAKE: injected)` note.

File: src/acp/client.ts [hunk 3]
```diff
@@ -423,6 +546,54 @@ export class WeChatAcpClient implements acp.Client {
...
+      return;
+    }
+
+    const mime = resource.mimeType?.trim() ?? "";
+    if (mime.toLowerCase().startsWith("image/")) {
+      // Route through the image pipeline so an image handed back as an
+      // embedded resource behaves exactly like an image content block.
+      await this.maybeSendImage({ data: resource.blob, mimeType: mime }, turn);
+      return;
+    }
```
[the remark this answers is not in the document]
Comment (Eskibear) [reply src/acp/client.ts]: Valid, fixed in 0513405. The blob path sanitizes `resource.mimeType` once up front; the sanitized value feeds the `image/*` routing check, the value forwarded to `maybeSendImage`, the placeholder line, and the log. A sanitized-but-bogus image mime (e.g. `image/png FAKE: injected`) simply fails the exact-match allow-list in `maybeSendImage` and is skipped, so nothing unsafe reaches the send path. Regression test covers CRLF injection in the placeholder.

File: src/acp/client.ts [hunk 4]
```diff
@@ -423,6 +571,56 @@ export class WeChatAcpClient implements acp.Client {
...
+
+    // Sanitized once here: the label feeds the placeholder, the logs, and
+    // the value forwarded to the image pipeline.
+    const mime = sanitizeInlineLabel(resource.mimeType ?? "");
+    if (mime.toLowerCase().startsWith("image/")) {
+      // Route through the image pipeline so an image handed back as an
+      // embedded resource behaves exactly like an image content block.
+      await this.maybeSendImage({ data: resource.blob, mimeType: mime }, turn);
+      return;
+    }
```
[the remark this answers is not in the document]
Comment (Eskibear) [reply src/acp/client.ts:777]: Valid, fixed in 5d11f08. The routing check now strips MIME parameters and requires the base type to be in the image allow-list before handing off to `maybeSendImage` (which then delivers the parameterized case as a normal image instead of skipping it); any other blob falls through to the visible placeholder. Regression tests cover `image/PNG; charset=binary` delivering as `image/png` and `image/tiff` falling back to the placeholder.

File: src/acp/client.ts [hunk 5]
```diff
@@ -29,6 +29,149 @@ const SUPPORTED_IMAGE_MIME_TYPES = new Set([
...
+ * are truncated with an explicit tail line inside the fence.
+ */
+function renderTextResource(name: string, mimeType: string | null | undefined, text: string): string {
+  let body = text;
+  if (text.length > MAX_RESOURCE_TEXT_CHARS) {
+    body =
+      text.slice(0, MAX_RESOURCE_TEXT_CHARS) +
+      `\n... [truncated, ${text.length - MAX_RESOURCE_TEXT_CHARS} more chars]`;
+  }
+  body = body.replace(/\n$/, "");
```
[the remark this answers is not in the document]
Comment (Eskibear) [reply src/acp/client.ts]: Valid, fixed in 5d11f08. The trailing strip now removes one CRLF, LF, or lone CR, so a Windows-style body no longer leaves a stray carriage return before the closing fence. Regression test asserts the fence sits directly after the last line for CRLF input.

File: src/acp/client.ts [hunk 6]
```diff
@@ -29,6 +29,149 @@ const SUPPORTED_IMAGE_MIME_TYPES = new Set([
...
+function resourceDisplayName(uri: string): string {
+  const trimmed = uri.trim();
+  if (!trimmed) return "resource";
+  const stripped = trimmed.replace(/[?#].*$/, "");
+  const segments = stripped.split("/").filter((s) => s.length > 0);
+  const last = segments[segments.length - 1] ?? "";
+  let name: string;
+  // A lone scheme-bearing segment is not a useful name.
+  if (!last || (segments.length === 1 && last.includes(":"))) {
+    name = trimmed;
```
[the remark this answers is not in the document]
Comment (Eskibear) [reply src/acp/client.ts:145]: Valid, fixed in b25cdf7. The scheme and authority are parsed off before segment extraction, so `https://example.com` (and the trailing-slash variant) now falls back to the full URI as documented, and opaque URIs like `untitled:Untitled-1` are handled explicitly instead of via the old lone-segment heuristic. Regression test covers both authority-only forms and asserts the bare host is never shown.

File: src/acp/client.ts [hunk 7]
```diff
@@ -29,6 +29,149 @@ const SUPPORTED_IMAGE_MIME_TYPES = new Set([
...
+  return sanitizeInlineLabel(name) || "resource";
+}
+
+function resourceFenceLanguage(name: string, mimeType?: string | null): string {
+  const mime = mimeType?.trim().toLowerCase() ?? "";
+  // Object.hasOwn, not `in`: prototype-chain keys like "toString" must not
+  // resolve to a language.
+  if (mime && Object.hasOwn(RESOURCE_MIME_LANGUAGES, mime)) {
+    return RESOURCE_MIME_LANGUAGES[mime];
+  }
```
[the remark this answers is not in the document]
Comment (Eskibear) [reply src/acp/client.ts:164]: Valid, fixed in b25cdf7. The lookup now strips MIME parameters before consulting the map, same normalization as the blob routing path. Regression test: `application/json; charset=utf-8` renders with a `json` hint.

File: src/acp/client.ts [hunk 8]
```diff
@@ -29,6 +29,149 @@ const SUPPORTED_IMAGE_MIME_TYPES = new Set([
...
+      text.slice(0, MAX_RESOURCE_TEXT_CHARS) +
+      `\n... [truncated, ${text.length - MAX_RESOURCE_TEXT_CHARS} more chars]`;
+  }
+  body = body.replace(/(\r\n|\n|\r)$/, "");
+  const longestRun = body.match(/`+/g)?.reduce((max, run) => Math.max(max, run.length), 0) ?? 0;
+  const fence = "`".repeat(Math.max(3, longestRun + 1));
+  const lang = resourceFenceLanguage(name, mimeType);
+  const mime = mimeType ? sanitizeInlineLabel(mimeType) : "";
+  const mimeNote = mime ? ` (${mime})` : "";
+  return `\n📎 ${name}${mimeNote}\n${fence}${lang}\n${body}\n${fence}\n`;
```
[the remark this answers is not in the document]
Comment (Eskibear) [reply src/acp/client.ts]: Valid, fixed in b25cdf7. Inner backtick runs of 8 or more are squeezed to 7 and the fence is capped at 8 ticks, so the fence always contains the body and the render stays bounded regardless of input. The squeeze mutates only the degenerate case (a run of 8+ backticks), which is a fair trade against an unbounded fence; normal code with inline triple-backtick runs still gets the exact longer-fence behavior. Regression test feeds a 3000-backtick run and asserts the fence caps at 8 with the render under one segment.

File: src/acp/client.ts [hunk 9]
```diff
@@ -423,6 +596,62 @@ export class WeChatAcpClient implements acp.Client {
...
+    // log-and-skip, which is fine for image content blocks but would
+    // silently drop a resource; anything else falls through to the
+    // visible placeholder below.
+    const baseMime = mime.split(";")[0].trim().toLowerCase();
+    if (SUPPORTED_IMAGE_MIME_TYPES.has(baseMime)) {
+      // Route through the image pipeline so an image handed back as an
+      // embedded resource behaves exactly like an image content block.
+      await this.maybeSendImage({ data: resource.blob, mimeType: baseMime }, turn);
+      return;
+    }
```
[the remark this answers is not in the document]
Comment (Eskibear) [reply src/acp/client.ts:777]: Valid, fixed in 405b036. The image-pipeline route now requires the image to be actually deliverable: supported base type, `showImages` not disabled, and an image sink configured. Anything else falls through to the visible `📎 [resource: ...]` placeholder, which also sets `producedMessage` so a resource-only turn no longer trips the empty-turn notice. Note the placeholder intentionally wins over the `showImages=false` silence: `showResources` is the knob that governs resource visibility, and the user chose to keep resources on. Regression tests cover both the `showImages=false` and no-sink paths.

File: src/acp/client.ts [hunk 10]
```diff
@@ -29,6 +29,174 @@ const SUPPORTED_IMAGE_MIME_TYPES = new Set([
...
+ * Render a text resource as a fenced code block with a one-line header
+ * naming it. The fence is longer than any backtick run inside the body so
+ * the content cannot break out, capped at {@link MAX_FENCE_TICKS} (longer
+ * inner runs are squeezed to fit); bodies above
+ * {@link MAX_RESOURCE_TEXT_CHARS} are truncated with an explicit tail line
+ * inside the fence.
+ */
+function renderTextResource(name: string, mimeType: string | null | undefined, text: string): string {
+  let body = text;
+  if (text.length > MAX_RESOURCE_TEXT_CHARS) {
```
Comment (formulahendry) [inline src/acp/client.ts]: The 4,000-character cap only applies to the resource body, so the header, fences, MIME note, and truncation suffix push the rendered block beyond `TEXT_CHUNK_LIMIT`. `splitText()` then prefers the newline after the opening fence and can send the header/opening fence, body, and closing fence as separate WeChat messages. Please calculate the body budget from the complete rendered overhead (or share the actual segment limit) and add a test asserting the final block is at most 4,000 characters / one `splitText()` segment.

File: src/acp/client.ts [hunk 11]
```diff
@@ -236,6 +407,8 @@ export class WeChatAcpClient implements acp.Client {
             } else if (c.type === "content" && c.content.type === "image") {
               imageContentBlocks++;
               await this.maybeSendImage(c.content, turn);
+            } else if (c.type === "content" && c.content.type === "resource") {
```
Comment (formulahendry) [inline src/acp/client.ts:468]: A successfully delivered image resource does not increment `imageContentBlocks`, so the `rawOutput.binaryResultsForLlm` fallback below still runs. If an agent populates both the standard resource block and the compatibility field with the same image, it will be delivered twice. Please have `maybeRenderResource()` report whether it sent an image (or otherwise track that result) and suppress the fallback when it did; a regression test with both fields populated would cover this.

**Review by formulahendry — CHANGES_REQUESTED** [review]
Thanks for the thorough implementation and strong edge-case coverage. Two correctness issues remain before merge: text resources can exceed the promised single WeChat segment and split their fences across messages, and image resources can be delivered twice when the standard resource block and the `rawOutput` compatibility field coexist. Please address the inline comments.

File: src/acp/client.ts (hunk 10 above)
Comment (Eskibear) [reply src/acp/client.ts]: Good catch, the cap only bounded the body while the header, fences, and language hint pushed the rendered block past `TEXT_CHUNK_LIMIT`, so `splitText` could cut mid-fence. Fixed in 4064e3f: `renderTextResource` now budgets the whole rendered block against the 4000-char segment limit (renamed the const to `MAX_RESOURCE_BLOCK_CHARS`). It computes the fixed overhead first (header with sanitized name and MIME, worst-case 8-tick fences, language hint) and truncates the body to the remaining budget, reserving space for the truncation tail. Since the sanitizer caps name and MIME at 120 chars each, the body budget never drops below roughly 3700.

The regression test renders an oversized resource and asserts the flushed block is at most 4000 chars and `splitText(block, 4000)` returns exactly one segment, plus a worst-case variant with a 300-char name and parameterized MIME. The closing fence is also asserted to survive truncation.

File: src/acp/client.ts (hunk 11 above)
Comment (Eskibear) [reply src/acp/client.ts:468]: Right, the resource route bypassed the `imageContentBlocks` counter, so the `rawOutput.binaryResultsForLlm` fallback would fire again for the same image. Fixed in 4064e3f: `maybeRenderResource` now returns true when it routes a blob into the image pipeline, the dispatch loop counts that into `resourceImages`, and the fallback gate is now `imageContentBlocks === 0 && resourceImages === 0`. The `[images: N content block]` log note also includes resource-delivered images.

Added the regression test you suggested: one completed `tool_call_update` carrying both an `image/png` resource content entry and the same image in `rawOutput.binaryResultsForLlm` asserts the image sink is called exactly once.

File: src/acp/client.ts [hunk 12]
```diff
@@ -29,6 +29,185 @@ const SUPPORTED_IMAGE_MIME_TYPES = new Set([
...
+ * either can carry newlines or control characters that would break the
+ * one-line format and inject extra lines into the chat transcript.
+ */
+function sanitizeInlineLabel(value: string): string {
+  const collapsed = value
+    .replace(/[\u0000-\u001f\u007f-\u009f\u2028\u2029]+/g, " ")
+    .replace(/\s+/g, " ")
+    .trim();
+  return collapsed.length > MAX_LABEL_CHARS ? `${collapsed.slice(0, MAX_LABEL_CHARS)}...` : collapsed;
+}
```
[the remark this answers is not in the document]
Comment (Eskibear) [reply src/acp/client.ts:116]: Valid, the slice-then-append meant labels could run 3 chars over the documented bound. Fixed in bf528a3: the cap now includes the ellipsis, so the returned label never exceeds `MAX_LABEL_CHARS`. No impact on the segment budget (it measures the actual rendered header), but the code now matches its contract. The worst-case overhead test asserts both header labels stay within 120 chars.

File: src/acp/client.ts [hunk 13]
```diff
@@ -29,6 +29,188 @@ const SUPPORTED_IMAGE_MIME_TYPES = new Set([
...
 const MAX_IMAGE_BYTES = 10 * 1024 * 1024;
 
+/**
+ * Budget for a rendered text-resource block: one WeChat segment
+ * (TEXT_CHUNK_LIMIT in bridge.ts). The whole block (header, fences,
+ * language hint, body, truncation tail) must fit, so `splitText()` never
+ * has to split mid-fence; the body budget is derived from this after
+ * subtracting the rendered overhead.
+ */
+const MAX_RESOURCE_BLOCK_CHARS = 4000;
```
[the remark this answers is not in the document]
Comment (Eskibear) [reply src/acp/client.ts]: Agreed on the drift risk. Fixed in 502bbc5: the limit now lives as an exported `TEXT_CHUNK_LIMIT` next to `splitText` in `weixin/send.ts` (the natural owner, and it avoids a bridge -> session -> client import cycle), `bridge.ts` drops its local copy, and `MAX_RESOURCE_BLOCK_CHARS` is defined as that constant. One source of truth for both the segmenter and the renderer budget.

File: src/acp/client.ts [hunk 14]
```diff
@@ -423,6 +614,72 @@ export class WeChatAcpClient implements acp.Client {
...
+    if ("text" in resource) {
+      if (!resource.text.trim()) {
+        opts.log(`[resource] skipped empty text resource: ${name}`);
+        return false;
+      }
+      turn.chunks.push(renderTextResource(name, resource.mimeType, resource.text));
+      turn.producedMessage = true;
+      opts.log(`[resource] rendered text resource ${name} (${resource.text.length} chars)`);
+      return false;
+    }
```
[the remark this answers is not in the document]
Comment (Eskibear) [reply src/acp/client.ts:755]: Valid, buffered narrative ahead of the block could push the combined text past the segment limit and `splitText` picks the last newline before the cut, which can land inside the fence. Fixed in 502bbc5: `maybeRenderResource` now flushes the buffer first when the combined length would exceed `TEXT_CHUNK_LIMIT`, so the block starts a fresh segment; a small narrative plus block still share one message (no premature flush). If that flush fails it re-buffers, trading the no-split guarantee for content preservation in that degraded path, same as the image and audio pipelines.

Narrative appended after the block is safe without further work: the block ends with its own newline, so a later greedy split always breaks at or after the closing fence. Two regression tests cover the flush-first case and the shared-segment case.

**Review by formulahendry — APPROVED** [review]
The previously requested changes have been addressed with focused regression coverage. The resource block now respects the shared WeChat segment limit, buffered narrative is flushed when necessary, and image resources correctly suppress duplicate rawOutput delivery. Looks good to merge.
